### ISSUE TITLE
update ODM v2 schema to the latest version

### ISSUE THREAD

**Issue by zargot** [Big-Life-Lab/PHES-ODM-Validation#233]
also, refactor code compensating for the previous inconsistency of having both lower and upper-case booleans in the ODM. See #235.

**Comment by zargot** [comment]
@yulric, are we gonna have all v2 revisions, or just the latest v2 schema?

**Comment by yulric** [comment]
I think we need to have all of them because every minor point version is a breaking change. @mathew-thomson can you confirm?

**Comment by zargot** [comment]
> I think we need to have all of them because every minor point version is a breaking change. @mathew-thomson can you confirm?

Then we'd potentially need different code-paths for every minor version. Is that what we want?

**Comment by mathew-thomson** [comment]
There shouldn't be too much in the way of breaking changes for each version. The upcoming release will have a few more though. 2.0 vs 2.1 vs 2.2.x should be mostly the same... Is there a way to easily test for differences that you can think of? I can devote some time to finding the changes and seeing where the breaking changes are (if they exist). There would be fewer than 10 (my guess) between the versions if there are any.

**Comment by zargot** [comment]
> There shouldn't be too much in the way of breaking changes for each version. The upcoming release will have a few more though. 2.0 vs 2.1 vs 2.2.x should be mostly the same... Is there a way to easily test for differences that you can think of? I can devote some time to finding the changes and seeing where the breaking changes are (if they exist). There would be fewer than 10 (my guess) between the versions if there are any.

It would be good if you could write a changelog for it. It would be useful for users as well.

**Comment by mathew-thomson** [comment]
I know……. I’ve been hoping to avoid it for too long. I’ll add it to my semi-urgent task list 😪😪😪

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by yulric** [comment]
> > I think we need to have all of them because every minor point version is a breaking change. @mathew-thomson can you confirm?
> 
> Then we'd potentially need different code-paths for every minor version. Is that what we want?

This would only be an issue if the breaking changes change the way that the library needs to parse the dictionary right? 

In any case, skipping an ODM version is fine but we should make sure that from one version to the next the validation information has not changed for example the set of allowed categories, min/max etc.

**Comment by zargot** [comment]
> > > I think we need to have all of them because every minor point version is a breaking change. @mathew-thomson can you confirm?
> > 
> > 
> > Then we'd potentially need different code-paths for every minor version. Is that what we want?
> 
> This would only be an issue if the breaking changes change the way that the library needs to parse the dictionary right?

Correct. The lower-case boolean parts is an example of how we would need to normalize the data at certain points depending on the version. It's probably fine, just a bit more complexity. 

> In any case, skipping an ODM version is fine but we should make sure that from one version to the next the validation information has not changed for example the set of allowed categories, min/max etc.

I'm not sure what you mean here, but we'll know what needs to be done between versions once we have the ODM changelog.

**Comment by mathew-thomson** [comment]
In any event,  I’ve added making these change logs to my to-do list and should have something for y’all before the end of the week. Thank you for your patience!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
